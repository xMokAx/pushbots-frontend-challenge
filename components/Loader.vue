<template>
  <v-overlay :value="isLoading" light color="rgba(255, 255, 255, 0.7)">
    <v-progress-circular indeterminate color="primary"></v-progress-circular>
  </v-overlay>
</template>

<script>
export default {
  props: {
    isLoading: {
      type: Boolean,
      required: true
    }
  }
}
</script>
