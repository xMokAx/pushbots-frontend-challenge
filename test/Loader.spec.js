import Vue from 'vue'
import Vuetify from 'vuetify'
import { mount } from '@vue/test-utils'
import Loader from '../components/Loader'

Vue.use(Vuetify)

const propsData = {
  isLoading: true
}

describe('Loader', () => {
  const wrapper = mount(Loader, {
    propsData
  })

  it('Should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
