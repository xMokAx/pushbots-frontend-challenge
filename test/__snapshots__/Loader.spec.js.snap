// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should match snapshot 1`] = `"<div class=\\"v-overlay v-overlay--active theme--dark\\" style=\\"z-index: 5;\\"><div class=\\"v-overlay__scrim\\" style=\\"opacity: 0.46; background-color: rgba(255, 255, 255, 0.7);\\"></div><div class=\\"v-overlay__content\\"><div role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"v-progress-circular v-progress-circular--indeterminate primary--text\\" style=\\"height: 32px; width: 32px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715\\" style=\\"transform: rotate(0deg);\\"><circle fill=\\"transparent\\" cx=\\"45.714285714285715\\" cy=\\"45.714285714285715\\" r=\\"20\\" stroke-width=\\"5.714285714285714\\" stroke-dasharray=\\"125.664\\" stroke-dashoffset=\\"125.66370614359172px\\" class=\\"v-progress-circular__overlay\\"></circle></svg><div class=\\"v-progress-circular__info\\"></div></div></div></div>"`;
